framework/physics/physics_material/mgxs/mgxs_combined.cc

```diff
@@ -0,0 +1,193 @@
+    sigma_f_.assign(n_grps, 0.0);
+    nu_sigma_f_.assign(n_grps, 0.0);
+    production_matrix_.assign(num_groups_, std::vector<double>(num_groups_, 0.0));
+
+    // Init prompt/delayed fission data
+    if (n_precs > 0)
+    {
+      nu_prompt_sigma_f_.assign(n_grps, 0.0);
+      nu_delayed_sigma_f_.assign(n_grps, 0.0);
+      precursors_.resize(n_precs);
+    }
+  }
+
+  // Combine the data
+  size_t precursor_count = 0;
+  for (size_t x = 0; x < xsecs.size(); ++x)
+  {
+    // Fraction of fissile density
+    const auto N_i = xsecs[x]->ScalingFactor();
+    const auto ff_i = xsecs[x]->IsFissionable() ? N_i / Nf_total : 0.0;
+
+    // Combine cross sections
+    const auto& sig_t = xsecs[x]->SigmaTotal();
+    const auto& sig_a = xsecs[x]->SigmaAbsorption();
+    const auto& sig_f = xsecs[x]->SigmaFission();
+    const auto& nu_p_sig_f = xsecs[x]->NuPromptSigmaF();
+    const auto& nu_d_sig_f = xsecs[x]->NuDelayedSigmaF();
+    const auto& F = xsecs[x]->ProductionMatrix();
+
+    // Here, raw cross sections are scaled by densities and spectra by
+    // fractional densities. The latter is done to preserve a unit spectra.
+    for (size_t g = 0; g < n_grps; ++g)
+    {
+      sigma_t_[g] += sig_t[g];
+      sigma_a_[g] += sig_a[g];
+
+      if (xsecs[x]->IsFissionable())
+      {
+        sigma_f_[g] += sig_f[g];
+        nu_sigma_f_[g] += sig_f[g];
+        for (size_t gp = 0; gp < num_groups_; ++gp)
+          production_matrix_[g][gp] += F[g][gp];
+
+        if (n_precs > 0)
+        {
+          nu_prompt_sigma_f_[g] += nu_p_sig_f[g];
+          nu_delayed_sigma_f_[g] += nu_d_sig_f[g];
+        }
+      }
+    } // for g
+
+    // Combine precursor data
+    // Here, all precursors across all materials are stored. The decay constants and delayed
+    // spectrum are what they are, however, some special treatment must be given to the yields.
+    // Because the yield dictates what fraction of delayed neutrons are produced from a
+    // given family, the sum over all families must yield unity. To achieve this end, all
+    // precursor yields must be scaled based on the fraction of the total density of materials
+    // with precursors they make up.
+
+    if (xsecs[x]->NumPrecursors() > 0)
+    {
+      const auto& precursors = xsecs[x]->Precursors();
+      for (size_t j = 0; j < xsecs[x]->NumPrecursors(); ++j)
+      {
+        size_t count = precursor_count + j;
+        const auto& precursor = precursors[j];
+        precursors_[count].decay_constant = precursor.decay_constant;
+        precursors_[count].fractional_yield = precursor.fractional_yield * ff_i;
+        precursors_[count].emission_spectrum = precursor.emission_spectrum;
+      } // for j
+
+      precursor_count += xsecs[x]->NumPrecursors();
+    }
+
+    // Set inverse velocity data
+    if (x == 0 and xsecs[x]->InverseVelocity().empty())
+      inv_velocity_ = xsecs[x]->InverseVelocity();
+    OpenSnLogicalErrorIf(
+      xsecs[x]->InverseVelocity() != inv_velocity_,
```

Comment:
are we looking for equality between two reals? no tiny fuzzy = ?

Reply:
Yep, it certainly looks like it. Not very robust. I'm really beginning to wonder if this code has ever been used.